### PR TITLE
chore: Undo bumping TableSource execution step version

### PR DESCRIPTION
**Pull request by Sullivan-Patrick**
### Description 
As part of [klip-50](https://github.com/confluentinc/ksql/blob/master/design-proposals/klip-50-partition-and-offset-in-ksqldb.md), to allow for joins involving tables to work with partition and offset, it is necessary to materialize partition and offset (see https://github.com/confluentinc/ksql/pull/7990). However, as we currently [do not support joins on windowed tables anyway](https://github.com/confluentinc/ksql/issues/8055), and have no plans to in the near future, the extra code complexity from introducing new steps is probably worse than not having the logic ready to go for if and when we do support these in the future.

This PR removes the execution step bump from the above PR.

### Testing done 
Fixed unit tests to expect new (old?) behavior.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java [hunk 1]
```diff
@@ -148,6 +150,70 @@ public static SourceBuilderV1 instance() {
...
+        valueSerde,
+        AutoOffsetReset.EARLIEST,
+        buildContext,
+        consumedFactory
+    );
+
+    final String stateStoreName = tableChangeLogOpName(source.getProperties());
+
+    final Materialized<Windowed<GenericKey>, GenericRow, KeyValueStore<Bytes, byte[]>> materialized
+        = buildWindowedTableMaterialized(
```
Comment (vcrfxia) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java:191]: Let's inline this method -- the method definition is longer than the implementation.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java [hunk 2]
```diff
@@ -148,6 +150,70 @@ public static SourceBuilderV1 instance() {
...
+        materializedFactory,
+        keySerde,
+        valueSerde,
+        stateStoreName
+    );
+
+    final KTable<Windowed<GenericKey>, GenericRow> ktable = buildWindowedKTable(
```
Comment (vcrfxia) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java:198]: As above, let's inline this method.

File: ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKSourceFactory.java [hunk 3]
```diff
@@ -139,26 +139,14 @@ private SchemaKSourceFactory() {
...
-
-    } else {
-      step = ExecutionStepFactory.tableSourceWindowedV1(
-          contextStacker,
-          dataSource.getSchema(),
-          dataSource.getKafkaTopicName(),
-          Formats.from(dataSource.getKsqlTopic()),
-          windowInfo,
-          dataSource.getTimestampColumn()
+    step = ExecutionStepFactory.tableSourceWindowed(
```
Comment (vcrfxia) [inline ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKSourceFactory.java:142]: Merge this into the line above.

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java [hunk 4]
```diff
@@ -176,65 +157,18 @@ public static SourceBuilder instance() {
...
-    final boolean windowed = streamSource instanceof WindowedTableSource;
-
     final FormatInfo formatInfo = streamSource.getFormats().getKeyFormat();
     final SerdeFeatures serdeFeatures = streamSource.getFormats().getKeyFeatures();
 
-    final KeyFormat keyFormat = windowed
-        ? KeyFormat.windowed(
-        formatInfo, serdeFeatures, ((WindowedTableSource) streamSource).getWindowInfo())
-        : KeyFormat.nonWindowed(formatInfo, serdeFeatures);
+    final KeyFormat keyFormat = KeyFormat.nonWindowed(formatInfo, serdeFeatures);
```
Comment (vcrfxia) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java:166]: nit: inline this in the line below.

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java [hunk 5]
```diff
@@ -1055,7 +1051,8 @@ private void givenWindowedSourceStream(final int pseudoColumnVersion) {
         OptionalInt.of(pseudoColumnVersion)
     );
   }
-  private void givenWindowedSourceStream() {
+  private void
+  givenWindowedSourceStream() {
```
Comment (vcrfxia) [inline ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java:1055]: nit: whitespace change.

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java [hunk 6]
```diff
@@ -1108,7 +1105,12 @@ private void givenWindowedSourceTableV1(final int pseudoColumnVersion) {
     );
   }
 
-  private void givenUnwindowedSourceTableV1(
+  private void givenWindowedSourceTable() {
+    givenWindowedSourceTable(ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION);
+  }
+
+
+    private void givenUnwindowedSourceTableV1(
```
Comment (vcrfxia) [inline ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java:1113]: nit: whitespace change.

**Review by vcrfxia — COMMENTED** [review]
Thanks @Sullivan-Patrick ! LGTM besides some nits inline.

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java (hunk 6 above)
Comment (vcrfxia) [reply ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java:1113]: I was actually referring to the fact that the indentation of this line is off but that works too 😆

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java (hunk 5 above)
Comment (vcrfxia) [reply ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java:1056]: I was actually referring to the fact that this line was split between two lines but your other change is good too 😆

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java [hunk 7]
```diff
@@ -188,14 +188,13 @@ public static SourceBuilderV1 instance() {
...
     final Materialized<Windowed<GenericKey>, GenericRow, KeyValueStore<Bytes, byte[]>> materialized
-        = buildWindowedTableMaterialized(
-        materializedFactory,
+        = materializedFactory.create(
         keySerde,
         valueSerde,
         stateStoreName
     );
 
-    final KTable<Windowed<GenericKey>, GenericRow> ktable = buildWindowedKTable(
```
Comment (vcrfxia) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java]: Delete the method definition for this now that it's no longer needed?

File: ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java (hunk 5 above)
Comment (Sullivan-Patrick) [reply ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java:1056]: I don't understand how I missed this one o.O

**Review by vcrfxia — APPROVED** [review]
LGTM! Will merge once the build passes.

**Pull request merged**